docs/usage/topology_aware_routing.md

```diff
@@ -0,0 +1,96 @@
+
+##### kube-proxy
+
+By default, with kube-proxy running in `iptables` mode, traffic is distributed randomly across all endpoints, regardless of where it originates from. In a cluster with 3 zones, traffic is more likely to go to another zone than to stay in the current zone.
+With the `TopologyAwareRouting` feature, kube-proxy filters the endpoints it routes to based on the hints in the EndpointSlice resource. In most of the cases, kube-proxy will prefer the endpoint(s) in the same zone. For more details, see the [following section](https://kubernetes.io/docs/concepts/services-networking/topology-aware-hints/#implementation-kube-proxy).
+
+## Topology-aware Services in the Seed cluster
+
+##### etcd-main-client and etcd-events-client
+
+The `etcd-main-client` and `etcd-events-client` Services are topology-aware. They are consumed by the kube-apiserver.
+
+##### kube-apiserver
+
+The `kube-apiserver` Service is topology-aware. It is consumed by the controllers running in the Shoot control plane.
+
+##### gardener-resource-manager
+
+The `gardener-resource-manager` Service that is part of the Shoot control plane is topology-aware. The resource-manager serves webhooks and the Service is consumed by the kube-apiserver for the webhook communication.
+
+##### vpa-webhook
+
+The `vpa-webhook` Service that is part of the Shoot control plane is topology-aware. It is consumed by the kube-apiserver for the webhook communication.
+
+## Topology-aware Services in the Garden cluster
+
+TBD
+
+## How to make a Service topology-aware?
+
+To make a Service topology-aware, the following annotation and label has to be added to the Service:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.kubernetes.io/topology-aware-hints: "auto"
+  labels:
+    endpoint-slice-hints.resources.gardener.cloud/consider: "true"
+```
+
+The `service.kubernetes.io/topology-aware-hints=auto` annotation is needed for kube-proxy. One of the prerequisites on kube-proxy side for using topology-aware routing is the corresponding Service to be annotated with the `service.kubernetes.io/topology-aware-hints=auto`. For more details, see the following [kube-proxy function](https://github.com/kubernetes/kubernetes/blob/b46a3f887ca979b1a5d14fd39cb1af43e7e5d12d/pkg/proxy/topology.go#L140-L186).
+The `endpoint-slice-hints.resources.gardener.cloud/consider=true` label is needed for gardener-resource-manager to prevent the EndpointSlice hints mutating webhook from selecting all EndpointSlice resources but only the ones that are labeled with the consider label.
+
+The Gardener extensions can use this approach to make a Service they deploy topology-aware.
```

Comment:
Is there any reason why a service should not be made topology-aware?
I do not see any issue in starting with a focussed approach based on the important services. However, identifying all relevant services is quite an effort. Therefore, I would be interested in your thoughts about generally enabling it (at some later point)?

Reply:
The topic is being discussed in https://github.com/gardener/gardener/pull/7191#discussion_r1085032036.

Reply:
Fair point. Let's discuss it there.